app/controllers/api/providers_controller.rb

```diff
@@ -85,6 +85,15 @@ def import_vm_resource(type, id = nil, data = {})
       end
     end
 
+    def options
+      req_class_name = @req.to_hash[:fullpath]
+                           .split('?').last.split('&')                        # '?' for query, '&' connects parameters
+                           .select { |param| param.starts_with?("ems_type") } # 'parameters with "ems_type" name
+                           .first.split('=').last                             # choosing only the first occuracne.
+      ems_class = ExtManagementSystem.descendants.select { |ems| ems.ems_type == req_class_name }.first
```

Comment:
@jntullo I didn't know of `descendant_get` before, but I am not sure it will help here because it will require sending the full class name (ie `ManageIQ::Providers::Openshift::ContainerManager`) instead of the `ems_type` reported by the providers. I will change to use `params[:ems_type]`